### PR TITLE
Jenkins Setup for Clang-CUDA

### PR DESCRIPTION
**Pull request by fthaler**

### CODE REVIEW

**Comment by fthaler** [comment]
launch strgrid

**Comment by fthaler** [comment]
launch strgrid

**Comment by fthaler** [comment]
launch icgrid

**Comment by fthaler** [comment]
launch strgrid

**Comment by fthaler** [comment]
launch strgrid

**Comment by fthaler** [comment]
launch strgrid

**Comment by fthaler** [comment]
launch strgrid

**Comment by fthaler** [comment]
launch icgrid

**Comment by fthaler** [comment]
launch perftest

**Comment by fthaler** [comment]
launch mpistrgrid

**Comment by fthaler** [comment]
launch icgrid

**Comment by fthaler** [comment]
launch strgrid

File: regression/icosahedral/CMakeLists.txt [hunk 1]
```diff
@@ -70,6 +70,9 @@ if(GT_ENABLE_BACKEND_CUDA)
     foreach(srcfile IN LISTS SOURCES)
         add_executable( ${srcfile}_cuda ${srcfile}.cu  )
         target_link_libraries(${srcfile}_cuda regression_main GridToolsTestCUDA)
+        if (GT_CUDA_COMPILATION_TYPE MATCHES "Clang-CUDA" OR GT_CUDA_COMPILATION_TYPE MATCHES "HIPCC-AMDGPU")
```
Comment (lukasm91) [inline regression/icosahedral/CMakeLists.txt]: `STREQUAL`?

File: jenkins/envs/daint_clang_nvcc.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+source $(dirname "$BASH_SOURCE")/daint.sh
+
+module load PrgEnv-gnu
+module swap gcc/7.3.0
+
+module use /project/csstaff/fthaler/install/daint/spack/share/spack/modules/cray-cnl6-haswell
+module load llvm-9.0.0-gcc-7.3.0-3lmdwud
```
Comment (lukasm91) [inline jenkins/envs/daint_clang_nvcc.sh]:
```suggestion
module load /project/csstaff/fthaler/install/daint/spack/share/spack/modules/cray-cnl6-haswell/llvm-9.0.0-gcc-7.3.0-3lmdwud
```
and remove module use

File: jenkins/envs/daint_clang_nvcc.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+source $(dirname "$BASH_SOURCE")/daint.sh
+
+module load PrgEnv-gnu
+module swap gcc/7.3.0
```
Comment (lukasm91) [inline jenkins/envs/daint_clang_nvcc.sh]: can you add a comment why you load gcc? I guess because of gfortran?

Comment (fthaler) [reply jenkins/envs/daint_clang_nvcc.sh]: It’s actually not strictly needed when I remember correctly, we could also use the default gfortran. But further down, I also tell Clang to use the gcc 7.3.0 toolchain, so it probably makes sense for consistency…

**Comment by fthaler** [comment]
launch strgrid

**Comment by fthaler** [comment]
launch icgrid

**Comment by fthaler** [comment]
launch perftest

**Comment by fthaler** [comment]
launch mpistrgrid

**Comment by fthaler** [comment]
launch perftest

**Comment by fthaler** [comment]
launch perftest

**Comment by fthaler** [comment]
launch strgrid

**Comment by fthaler** [comment]
launch icgrid

**Comment by fthaler** [comment]
launch mpistrgrid

**Pull request merged**